snafu-derive/src/lib.rs

```diff
@@ -569,19 +581,31 @@ impl syn::parse::Parse for SnafuAttributeBody {
     }
 }
 
+struct DocComment(SnafuAttribute);
+
+impl syn::parse::Parse for DocComment {
+    fn parse(input: syn::parse::ParseStream) -> SynResult<Self> {
+        let _: Token![=] = input.parse()?;
+        let doc: syn::LitStr = input.parse()?;
+        Ok(DocComment(SnafuAttribute::DocComment(doc.value())))
+    }
+}
+
 fn attributes_from_syn(attrs: Vec<syn::Attribute>) -> MultiSynResult<Vec<SnafuAttribute>> {
     use syn::parse2;
 
     let mut ours = Vec::new();
     let mut errs = Vec::new();
 
-    let parsed_attrs = attrs
-        .into_iter()
-        .filter(|attr| attr.path.is_ident("snafu"))
-        .map(|attr| {
-            let body: SnafuAttributeBody = parse2(attr.tts)?;
-            Ok(body.0)
-        });
+    let parsed_attrs = attrs.into_iter().filter_map(|attr| {
```

Comment:
Could we change this to `flat_map`? I tend to prefer it as it’s strictly more powerful than `filter_map`.